### ISSUE TITLE
Rename column, but ThinkingSphinx::SphinxError: unknown column: 'shop_address_lookup'

### ISSUE THREAD

**Issue by victorngkp** [pat/thinking-sphinx#1008]
Using Rails 4, Thinking Sphinx 3.2.0. I recently renamed a column from for_search to address_lookup. The Shop model has the column address_lookup, and I have Country which is associated to Shop model.

Here's my Country index:

```
ThinkingSphinx::Index.define :state, with: :real_time do
  indexes name
  indexes shop_address_lookup
  ...
end
```
I removed the `development.sphinx.conf`, and ran `ts:regenerate`. Indexing Shop worked fine, but when it tried to index Country, it kept throwing the following error:

```
Generating index files for country_core
rake aborted!
ThinkingSphinx::SphinxError: unknown column: 'shop_address_lookup' - REPLACE INTO country_core (id, `sphinx_internal_class_name`, `name`, `country_status`, `shop_address_lookup`
...
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
Innertube::Pool::BadResource: Innertube::Pool::BadResource
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
ThinkingSphinx::QueryExecutionError: unknown column: 'shop_address_lookup'
...
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
Mysql2::Error: unknown column: 'shop_address_lookup'
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
/Users/abc/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
Tasks: TOP => ts:regenerate => ts:generate
(See full trace by running task with --trace)
```

I only renamed the column, and already updated my index file. What can I do to fix this?

Note:
- When I change the index from `shop_address_lookup` to `shop_for_search` again, without changing the actual column name, the index works fine.
- I tried to revert everything, so moving back to `for_address`, reverted the index, and now it doesn't work. TS is complaining `unknown column: 'shop_for_search'`.

**Comment by victorngkp** [comment]
Managed to get this to work finally - remove `db/sphinx/development/*` before running `rake ts:regenerate`.

**Comment by pat** [comment]
Great to know you got it figured out, but I'm actually going to re-open this issue - `ts:regenerate` should be removing all relevant files from `db/sphinx/RAILS_ENV`, and the fact that it isn't means there's a bug. I'll try to get it fixed sooner rather than later!

**Comment by pat** [comment]
Found the problem, fixed it in cf16d18285e57e72d79234ec37ac8a5aa301296b. Thanks for logging the issue!
